.ibm/pipelines/utils.sh

```diff
@@ -285,3 +287,441 @@ install_pipelines_operator() {
+  GITHUB_APP_CLIENT_SECRET=$(cat /tmp/secrets/GITHUB_APP_4_CLIENT_SECRET)
+
+  export GITHUB_APP_APP_ID
+  export GITHUB_APP_CLIENT_ID
+  export GITHUB_APP_PRIVATE_KEY
+  export GITHUB_APP_CLIENT_SECRET
+  echo "GitHub App 4 credentials set for current job."
+}
+
+apply_yaml_files() {
+  local dir=$1
+  local project=$2
+  echo "Applying YAML files to namespace ${project}"
+
+  oc config set-context --current --namespace="${project}"
+
+  local files=(
+      "$dir/resources/service_account/service-account-rhdh.yaml"
+      "$dir/resources/cluster_role_binding/cluster-role-binding-k8s.yaml"
+      "$dir/resources/cluster_role/cluster-role-k8s.yaml"
+      "$dir/resources/cluster_role/cluster-role-ocm.yaml"
+      "$dir/auth/secrets-rhdh-secrets.yaml"
+    )
+
+    for file in "${files[@]}"; do
+      sed -i "s/namespace:.*/namespace: ${project}/g" "$file"
+    done
+
+    for key in GITHUB_APP_APP_ID GITHUB_APP_CLIENT_ID GITHUB_APP_PRIVATE_KEY GITHUB_APP_CLIENT_SECRET GITHUB_APP_JANUS_TEST_APP_ID GITHUB_APP_JANUS_TEST_CLIENT_ID GITHUB_APP_JANUS_TEST_CLIENT_SECRET GITHUB_APP_JANUS_TEST_PRIVATE_KEY GITHUB_APP_WEBHOOK_URL GITHUB_APP_WEBHOOK_SECRET KEYCLOAK_CLIENT_SECRET ACR_SECRET GOOGLE_CLIENT_ID GOOGLE_CLIENT_SECRET K8S_CLUSTER_TOKEN_ENCODED OCM_CLUSTER_URL GITLAB_TOKEN; do
+      sed -i "s|${key}:.*|${key}: ${!key}|g" "$dir/auth/secrets-rhdh-secrets.yaml"
+    done
+
+    oc apply -f "$dir/resources/service_account/service-account-rhdh.yaml" --namespace="${project}"
+    oc apply -f "$dir/auth/service-account-rhdh-secret.yaml" --namespace="${project}"
+    oc apply -f "$dir/auth/secrets-rhdh-secrets.yaml" --namespace="${project}"
+
+    oc apply -f "$dir/resources/cluster_role/cluster-role-k8s.yaml" --namespace="${project}"
+    oc apply -f "$dir/resources/cluster_role_binding/cluster-role-binding-k8s.yaml" --namespace="${project}"
+    oc apply -f "$dir/resources/cluster_role/cluster-role-ocm.yaml" --namespace="${project}"
+    oc apply -f "$dir/resources/cluster_role_binding/cluster-role-binding-ocm.yaml" --namespace="${project}"
+
+    escaped_url=$(printf '%s\n' "${ENCODED_API_SERVER_URL}" | sed 's/[\/&]/\\&/g')
+    sed -i "s/K8S_CLUSTER_API_SERVER_URL:.*/K8S_CLUSTER_API_SERVER_URL: ${escaped_url}/g" "$dir/auth/secrets-rhdh-secrets.yaml" \
+      && echo "Updated K8S_CLUSTER_API_SERVER_URL in secrets file." \
+      || echo "Failed to update K8S_CLUSTER_API_SERVER_URL." >&2
+
+    sed -i "s/K8S_CLUSTER_NAME:.*/K8S_CLUSTER_NAME: ${ENCODED_CLUSTER_NAME}/g" "$dir/auth/secrets-rhdh-secrets.yaml"
+
+    token=$(oc get secret "${secret_name}" -n "${project}" -o=jsonpath='{.data.token}')
+    sed -i "s/OCM_CLUSTER_TOKEN: .*/OCM_CLUSTER_TOKEN: ${token}/" "$dir/auth/secrets-rhdh-secrets.yaml"
+
+    # Select the configuration file based on the namespace or job
+    config_file=$(select_config_map_file)
+    # Apply the ConfigMap with the correct file
+    create_app_config_map "$config_file" "$project"
```

Comment:
This has to be differentiated for k8s jobs (AKS, GKE). An `if` or `case` switch is needed here. 
For k8s jobs, use `create_app_config_map_k8s` (currently, it is named `create_app_configmap_aks_gke`, but I suggest to change it respectively).